Nos/Views/CircularFollowButton.swift

```diff
@@ -0,0 +1,65 @@
+import CoreData
+import Dependencies
+import Logger
+import SwiftUI
+
+/// A small, circular follow button that appears in the lower right corner of a user's avatar.
+/// Allows the current user to follow or unfollow the author,
+/// and updates its own appearance based on follow state.
+struct CircularFollowButton: View {
+    @ObservedObject var author: Author
+    @Environment(CurrentUser.self) private var currentUser
+    @Dependency(\.analytics) private var analytics
+    @Dependency(\.crashReporting) private var crashReporting
+
+    private let diameter: CGFloat = 30
+
+    var body: some View {
+        let following = currentUser.isFollowing(author: author)
+
+        Button {
+            Task {
```

Comment:
Can you disable the button while the Task is running? Like `ActionButton`. Maybe we need an `AsyncButton` that does this and makes no assumptions about styling (like `ActionButton` and `SecondaryActionButton`).

Reply:
@mplorentz I can, and I suppose I will, but I'll also point out that `ActionButton` has bugs that I wasn't able to fix when I worked on a different bug. Basically, `ActionButton` does not guarantee that it's disabled while its Task is running. But I suppose it's still worth trying here.

Reply:
Done!